### PR TITLE
enhancement - allow multiple shared inputs

### PR DESCRIPTION
**Pull request by joefitter**
- cast shared to array, reverse and loop over, extending from right to left.

### CODE REVIEW

**Comment by joefitter** [comment]
@easternbloc @JoeChapman @gxxm @daniel-ac-martin @danielepolencic could you have a look please guys?

**Comment by joefitter** [comment]
^ not sure why this is even building?

**Comment by JoeChapman** [comment]
Must've been enabled in travis. Maybe add a travis file.

**Comment by joefitter** [comment]
That will be out of scope for this PR - I'll add an issue to add a travis config and linting, and TESTS

**Comment by joefitter** [comment]
disabled in travis until https://github.com/UKHomeOffice/hof-transpiler/issues/16 is resolved

**Comment by joefitter** [comment]
@easternbloc @JoeChapman @danielepolencic @daniel-ac-martin are you guys happy with this?

File: bin/hof-transpiler [hunk 1]
```diff
@@ -84,22 +91,24 @@ _.each(sources, function (src) {
...
-        var streamLang = getLanguage(path.parse(writerStream.path).dir);
-        var parsedStream = null;
-        var stringStream = '';
+      shared.forEach(function (sharedSource) {
+        walk.walk(sharedSource).on('file', function (root, fileStats, next) {
+          var lang = getLanguage(root);
+          var fileName = fileStats.name.replace('.json', '');
+          var streamLang = getLanguage(path.parse(writerStream.path).dir);
+          var parsedStream = null;
+          var stringStream = '';
```
Comment (JoeChapman) [inline bin/hof-transpiler:100]: could use const and let for these

File: bin/hof-transpiler [hunk 2]
```diff
@@ -13,13 +13,20 @@ var sources = argv._;
...
+
 if (!sources.length) {
   throw new Error('No src specified');
 }
 
-var sindex = sources.indexOf(shared);
-if (writeShared !== true && sindex > -1) {
-  sources.splice(sindex, 1);
+if (writeShared !== true) {
+  sources = sources.filter(function (source) {
```
Comment (JoeChapman) [inline bin/hof-transpiler:27]: arrow func?

File: bin/hof-transpiler [hunk 3]
```diff
@@ -84,22 +91,24 @@ _.each(sources, function (src) {
...
       var sharedMap = {};
 
-      walk.walk(shared).on('file', function (root, fileStats, next) {
-        var lang = getLanguage(root);
-        var fileName = fileStats.name.replace('.json', '');
-        var streamLang = getLanguage(path.parse(writerStream.path).dir);
-        var parsedStream = null;
-        var stringStream = '';
+      shared.forEach(function (sharedSource) {
+        walk.walk(sharedSource).on('file', function (root, fileStats, next) {
```
Comment (JoeChapman) [inline bin/hof-transpiler:95]: arr func?

File: bin/hof-transpiler (hunk 1 above)
Comment (joefitter) [inline bin/hof-transpiler:100]: That would be a separate issue, not in the scope of this. Having a half es5 half es6 file is never a good thing

File: bin/hof-transpiler (hunk 2 above)
Comment (joefitter) [inline bin/hof-transpiler:27]: as above

File: bin/hof-transpiler (hunk 3 above)
Comment (joefitter) [inline bin/hof-transpiler:95]: as above

**Comment by joefitter** [comment]
@JoeChapman https://github.com/UKHomeOffice/hof-transpiler/issues/21

File: bin/hof-transpiler (hunk 1 above)
Comment (joefitter) [inline bin/hof-transpiler:100]: Added an issue to es6ify - https://github.com/UKHomeOffice/hof-transpiler/issues/21

**Pull request merged**